docs/architecture/domain.md

```diff
@@ -0,0 +1,117 @@
+## Definition Model
+
+![Definition](assets/definition-model.png)
```

Comment:
I agree with @d2lam, the graph is a little confusing. 
You could put numbers to indicate order of execution and/or put job names that indicate order (Job1, Job2, etc.)